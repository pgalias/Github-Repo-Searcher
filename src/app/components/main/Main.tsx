import React, { FunctionComponent, ReactElement } from 'react';
import styled from '@emotion/styled';
import Container from '@material-ui/core/Container';

const MainWrapper = styled.main`
  background-color: #f3f4f6;
`;

export const Main: FunctionComponent<{ children: ReactElement }> = ({ children }) => (
  <MainWrapper>
    <Container fixed>{children}</Container>
  </MainWrapper>
);
