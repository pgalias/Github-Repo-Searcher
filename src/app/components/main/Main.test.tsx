import React from 'react';
import { render, screen } from '@testing-library/react';
import { Main } from './Main';
import { Loader } from '../loader';

describe('Main component', () => {
  test('should render with text type children', () => {
    render(
      <Main>
        <p>foo</p>
      </Main>,
    );

    expect(screen.getByRole('main')).toMatchSnapshot();
  });

  test('should render with react component type children', () => {
    render(
      <Main>
        <Loader />
      </Main>,
    );

    expect(screen.getByRole('main')).toMatchSnapshot();
  });
});
