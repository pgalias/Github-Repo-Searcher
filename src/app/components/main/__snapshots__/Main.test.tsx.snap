// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render with react component type children 1`] = `
<main
  class="css-1vv3q56"
>
  <div
    class="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
  >
    <div
      aria-busy="true"
      class="css-q3dffh"
      role="alert"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</main>
`;

exports[`Main component should render with text type children 1`] = `
<main
  class="css-1vv3q56"
>
  <div
    class="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
  >
    <p>
      foo
    </p>
  </div>
</main>
`;
